tsdb/index/tsi1/log_file.go

```diff
@@ -1039,15 +1040,20 @@ func (e *LogEntry) UnmarshalBinary(data []byte) error {
 	// Parse series id.
 	if len(data) < 1 {
 		return io.ErrShortBuffer
+	} else if seriesID, n = binary.Uvarint(data); n == 0 {
+		return io.ErrShortBuffer
+	} else if n < 0 {
+		return fmt.Errorf("parsing binary-encoded uint64 value failed; binary.Uvarint() returned %d", n)
 	}
-	seriesID, n := binary.Uvarint(data)
 	e.SeriesID, data = uint64(seriesID), data[n:]
```

Comment:
Still need to check `n < len(data)`, or we could get an index out of range panic on this line.

Reply:
Done